config/passport-setup.js

```diff
@@ -0,0 +1,37 @@
+const passport = require('passport');
+const GoogleStrategy = require('passport-google-oauth20').Strategy;
+
+const Coaches = require('../coaches/coaches-model');
+
+
+passport.use( 
+  new GoogleStrategy( {
+    clientID: process.env.CLIENT_ID,
+    clientSecret:process.env.CLIENT_SECRET,
+    callbackURL:'/auth/google/callback'
+  }, (accessToken, refreshToken, users, done) => {
+    // console.log(users);
+    // //check if user already exists in our own db
+    Coaches
+      .findCoachBy(users.emails[0].value)
```

Comment:
I thing it might be better here to use `users._json.email` instead since an array of emails might not have their preferred/main one as the first index.